### PR TITLE
fix(root): #544 Add code typography tokens

### LINKED ISSUE #544: [ic-typography]: Add code typography tokens
## Summary of the bug
The code variants for `ic-typography` (seen below, taken from Figma) are not available in the `ic-ui-kit` repo.

Screenshot 2023-03-31 at 20 02 23

## Acceptance Criteria
1.  Tokens for the code variants should be listed in the `variables.css`
2. The code variants should be listed in the `IcTypographyVariants` type found in the `utils/types.ts` file (so it can be passed through the `variant` prop of `ic-typography`)
3. CSS rulesets should be added to `ic-typography.css` for the code variants
4. The code variants should be added to the "variants" & "vertical margins" stories in `ic-typography.stories.mdx`
5. Unit tests for the code variants should be added to `ic-typography.spec.ts`
6. Visual tests should be updated following the addition to the stories in `ic-typography.stories.mdx`

### PR DESCRIPTION
**Pull request by joshmu**
The code variants for **ic-typography** for the **ic-ui-kit**

fix #544

## Checklist
- [x] I have added relevant unit and visual regression tests.
- [x] I have manually accessibility tested any changes, if relevant.
- [x] I have ensured any changes match the Figma component library.

### CODE REVIEW

**Review by GCHQ-Developer-112 — CHANGES_REQUESTED** [review]
Thank you for your contribution ✨ 
Overall it looks good! 

I've left some comments on the files and have a few other comments:

1. Please can you do a rebase off the develop branch: `git rebase origin/develop`. This will remove the extra commits that have been pulled through. If this doesn't work, please reply to this comment and we can walk you through how to drop the commits using an interactive rebase.

2. If a PR involves changes to more than one package, there needs to be a commit for each package. In this case there are changes to `packages/react` and `packages/web-components`, so there should be 2 commits and your commit messages should look something like this:
    - fix(web-components): Add code typography tokens
    - fix(react): Add code typography variants to React Storybook

3. Please note: the issue number should not be in the short description of the commit message. The issue number comes into the commit message later. Commitzen prompts you to add the issue number with the question: _Does this change affect any open issues? (y/n)._

More information on how to structure your commit can be found in our [`ic-ui-kit` CONTRIBUTING.md](https://github.com/mi6/ic-ui-kit/blob/main/CONTRIBUTING.md#git-commit)

File: packages/web-components/src/global/variables.css
Comment (joshmu) [reply packages/web-components/src/global/variables.css:58]: No problem, I'll make the change, I typically would keep it as a singular value however I noticed this convention being done in the codebase.

eg:
```css
  --ic-font-line-height-subtitle-large: 1.5rem;
  --ic-font-line-height-subtitle-small: 1.5rem;
```

**Review by GCHQ-Developer-530 — DISMISSED** [review]

**Review by MI6-451 — APPROVED** [review]

**Review by GCHQ-Developer-112 — COMMENTED** [review]
Hey @joshmu, great job on this! 

At the moment, we can't merge this ticket in as our usual checks for linting, prettier, unit tests & end-to-end tests don't seem to be running on your commits. Not quite sure why this is.

Please could you follow the steps below. This should pull in the latest changes on develop and hopefully allow us to manually run our checks 
```shell
# Add the remote, call it "upstream":
## If you're using SSH
git remote add upstream git@github.com:mi6/ic-ui-kit.git

## If you're using HTTPS 
git remote add upstream https://github.com/mi6/ic-ui-kit.git

# Fetch all the branches of that remote into remote-tracking branches
git fetch upstream

# Make sure that you're on your develop branch:
git checkout develop

# Rewrite your develop branch so that any commits of yours that aren't already in upstream/develop are replayed on top of that other branch:
git rebase upstream/develop

# Switch on to the branch you're working on
git checkout <your-branch>

# Rewrite the branch you're working on so that any commits of yours that # aren't already in origin/develop are replayed on top of that 
# other branch: 
git rebase origin/develop

```

Happy to approve after this 😀

**Comment by joshmu** [comment]
Hey @GCHQ-Developer-112 

No problem, let's try again. :)

**Comment by GCHQ-Developer-112** [comment]
> Hey @GCHQ-Developer-112
> 
> No problem, let's try again. :)

Thank you so much for doing this! Unfortunately the checks still aren't running on your PR but luckily I can see the checks on your forked repo. Please could you run the following in your IDE:

- `npm run prettier:fix` to resolve the prettier error
- `npm run test-visual:update` to update your visual tests and only commit the typography visual test

**Comment by joshmu** [comment]
Hey @GCHQ-Developer-112
Updated as per your comment.
I have a suspicion this will still not work as the visual regression tests are failing in the pipeline on already updated images.
Could the chrome instance on my local not match with your pipeline?

**Comment by GCHQ-Developer-530** [comment]
> Hey @GCHQ-Developer-112 Updated as per your comment. I have a suspicion this will still not work as the visual regression tests are failing in the pipeline on already updated images. Could the chrome instance on my local not match with your pipeline?

We're aware of some issues with visual regression tests on windows devices, this could be the cause of this issue? We have a github action we can run on our branches to update the visual tests manually for such devices, but unsure if we can do it for a forked branch. Are you using a windows developer environment? If so we can try and find a way to run the github action on your branch to update the tests for you.

**Review by GCHQ-Developer-530 — DISMISSED** [review]

**Comment by GCHQ-Developer-112** [comment]
> Hey @GCHQ-Developer-112 Updated as per your comment. I have a suspicion this will still not work as the visual regression tests are failing in the pipeline on already updated images. Could the chrome instance on my local not match with your pipeline?

Hey @joshmu, I've cloned your repo and had a look at Storybook. It looks like the code variants are using the fallback font. I've had a look at how we implement Open Sans, our main font and it seems that we need to install and import the Source Code Pro font. Please could you follow the steps below:

In your IDE, please run the following commands:
```shell
cd packages/fonts
npm install @fontsource/source-code-pro --save-dev
```

Add the following to the fonts.js file (packages/fonts/src/fonts.js):
```js
// Source Code Pro
 import '@fontsource/source-code-pro/400.css';
import '@fontsource/source-code-pro/600.css';
```
Please can you also commit the changes to the README.md file in the ic-typography folder. The information in the README file is used in the prop tables on our [guidance site](https://design.sis.gov.uk). 

Lastly, please can you update the visual tests for ic-typography. There should be changes now that we're importing the Source Code Pro font.  If the visual tests still fail after this, we will merge this PR in and update the visual tests on our end.

Thank you for your work 😀

**Comment by joshmu** [comment]
Somehow missed a bit on my rebase, now it should work...

**Review by GCHQ-Developer-112 — APPROVED** [review]
Thank you so much for you work on this @joshmu! 🌟

**Review by GCHQ-Developer-530 — APPROVED** [review]

**Pull request merged**
